### ISSUE TITLE
Unable to overwrite `format_input` in `theme` trait

### ISSUE THREAD

**Issue by jorddn** [fadeevab/cliclack#67]
When wanting to customize the prompts further using the `Theme` trait, the following functions are unable to be overwritten due to them using `StringCursor` which is not exported anywhere:
```rs
// cannot find type `StringCursor` in this scope
fn cursor_with_style(&self, cursor: &StringCursor, new_style: &Style) -> String { ... }
fn format_input(&self, state: &ThemeState, cursor: &StringCursor) -> String { ... }
fn format_placeholder(&self, state: &ThemeState, cursor: &StringCursor) -> String { ... }
```

I'm not sure if it's possible without a change being made that re-exports `StringCursor`, (perhaps with `#[doc(hidden)]` if preferred) but if there's an alternate solution that requires no changes to the source I would appreciate it being shared!

**Comment by fadeevab** [comment]
@jorddn Could you double check whether customizing `input_style` would be enough for you? `format_input` is quite complex method to be modified.

**Comment by jorddn** [comment]
Currently this is the closest I can get to my desired result without modifying the source myself:
![Screenshot 2024-08-08 054616](https://github.com/user-attachments/assets/a5b291a4-d8bc-424b-8afc-d8d1d1359aee)

Ideally, this would be my output (where the spacing has been modified and the bar removed)
![Screenshot 2024-08-08 061306](https://github.com/user-attachments/assets/ba40f867-345d-429a-bbf3-5ead80703d78)

**Comment by fadeevab** [comment]
@jorddn I see what you're trying to achieve. Exposing the cursor type would be the easiest way to go.

**Comment by fadeevab** [comment]
@jorddn The fix: #70

**Comment by fadeevab** [comment]
@jorddn Published in `0.3.3`
